native/src/utils/renderHtml.ts

```diff
@@ -5,237 +5,259 @@ import { ERROR_MESSAGE_TYPE, getFontFaceSource, HEIGHT_MESSAGE_TYPE, WARNING_MES
+            '${WARNING_MESSAGE_TYPE}')
         }
-      } catch (e) {
-        reportError(e.message + 'occurred while decoding and looking for ' + item.src + ' in the dictionary',
-          '${WARNING_MESSAGE_TYPE}')
       }
-    }
-  })();
 
-  (function() {
-    var container = document.getElementById('measure-container')
+      for (var i = 0; i < srcs.length; i++) {
+        var item = srcs[i]
+        try {
+          var newResource = cacheDictionary[decodeURI(item.src)]
+          if (newResource) {
+            item.src = newResource
+          }
+        } catch (e) {
+          reportError(e.message + 'occurred while decoding and looking for ' + item.src + ' in the dictionary',
+            '${WARNING_MESSAGE_TYPE}')
+        }
+      }
+    })();
 
-    function adjustHeight () {
-      container.setAttribute('style', 'padding: 1px 0;'); // Used for measuring collapsed vertical margins
+    (function() {
+      var container = document.getElementById('measure-container')
 
-      if (!window.ReactNativeWebView) {
-        return window.setTimeout(adjustHeight, 100);
-      }
+      function adjustHeight () {
+        container.setAttribute('style', 'padding: 1px 0;'); // Used for measuring collapsed vertical margins
 
-      var height = container.getBoundingClientRect().height - 2
-      window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${HEIGHT_MESSAGE_TYPE}', height: height }));
-      container.setAttribute('style', '');
-    }
+        if (!window.ReactNativeWebView) {
+          return window.setTimeout(adjustHeight, 100);
+        }
 
-    window.addEventListener('load', adjustHeight);
-    window.addEventListener('resize', adjustHeight);
-    var details = document.querySelectorAll("details")
-    details.forEach(detail => detail.addEventListener("toggle", adjustHeight))
-  })();
-`
+        var height = container.getBoundingClientRect().height - 2
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${HEIGHT_MESSAGE_TYPE}', height: height }));
+        container.setAttribute('style', '');
+      }
 
+      window.addEventListener('load', adjustHeight);
+      window.addEventListener('resize', adjustHeight);
+      var details = document.querySelectorAll("details")
+      details.forEach(detail => detail.addEventListener("toggle", adjustHeight))
+    })();
+
+    // iframe handling - remove not allowed iframe src from DOM and add tracking parameter
+    (function() {
+      var iframes = document.querySelectorAll('iframe')
+      iframes.forEach((el) => {
+        // TODO 2588- replace by allowedIframeSources from buildConfig instead of hardcoding
+        var allowedIframeSources = ['vimeo']
+        if (allowedIframeSources.some(url => el.src.indexOf(url) > 0)) {
+          var url = new URL(el.src)
+          url.searchParams.append('dnt', '1')
+          el.setAttribute('src', url.href)
+      } else {
+          el.parentNode.removeChild(el)
+        }
+      })
+    })();
+  `
```

Comment:
Question (because I am actually not sure whether it would be easily possible or even if it would be possible if it would make sense): Could we directly strip the `<iframe>` tags from the html in the api-client already?

Then we would only need to do this in one place (but probably for all endpoints with html). So probably it would still not make sense.

Reply:
I mean we could add this `do not track` in the `api-client` but the rest of the `iframe` handling is different on `native` and `web` so even we need a separate handling.
Additionally i think we not just want to strip this but show a proper message that it was blocked to show the maintainer of the page that he/she is not allowed to use this source instead of just stripping it out.